core/src/main/java/nl/nn/adapterframework/lifecycle/VerifyDatabaseConnectionBean.java

```diff
@@ -54,6 +55,10 @@ public void afterPropertiesSet() throws Exception {
 			TransactionStatus status = transactionManager.getTransaction(new DefaultTransactionDefinition());
 
 			try (Connection connection = getDefaultDataSource().getConnection()) {
+				if(!connection.isValid(5)) {
```

Comment:
Ik ben bang dat dit nog steeds niet genoeg zal zijn voor DB2 totdat we een validation-query toevoegen die test dat tabel bestaat waarvan we verwachten dat die bestaat, zoals IBISSTORE.

(Die validation-query moet misschien in de driver properties worden geconfigureerd, misschien in de connection-pool properties, dat zie ik zo snel niet en connection pooling zit in een ander PR dus buiten scope hier).

Maar in ieder geval is dit een stap in de juiste richting.

Reply:
Ja, bij BTM was dit het geval (die doet als het goed is een isValid check) Narayana deed dat niet. Het nieuwe pooling mechanisme wel (#5142). Ik vind het ook vreemd dat de transactionmanager ook goed op kan starten, dat zou dan ook niet mogen. In onze tests gaat het zonder TX manager wel goed.

Ik heb gekeken naar de `DatabaseStartupValidator` maar die is sinds Spring 5.3 gedeprecate 'in favor of the JDBC 4.0 connection validation'. Spring boot heeft een package/enum (`DatabaseDriver`) met validation queries die we kunnen gebruiken. Maar ook die worden discouraged. 

Volgens de DB2 documenatie (jcc driver) hoort dat ding JDBC4 te implementen en volledig te ondersteunen.